### DISCUSSION TITLE
WSL2 Python segmentation fault

### DISCUSSION THREAD

**Discussion by ghost** [google/or-tools#2455] [Build and configuration issues] [answered]
I am running an optimization program on wsl2 kali distribution. The program works fine on Windows and other Python programs run fine on WSL2. 

However, with ortools, I get a segmentation error and the program crashes. 

Is there anyone who knows whether this is an ortools or wsl2 related issue?

**Comment by lperron** [comment]
OR-Tools is mostly C++, with a python wrapping.
I do not know the characteristics of the kali distribution.
Does it use the glibc ?

**Reply by ghost** [reply]
It does use glibc. I am guessing this might be related to how wsl2 virtual machine manages memory but cannot find anyone else having a similar problem.

Another detail is I have two separate VRP problems. The application always crashes after the first one finishes and the second one is about to begin.

**Reply by Mizux** [reply]
providing Python version and glibc version could help us too...

**Reply by ghost** [reply]
> providing Python version and glibc version could help us too...

I am on Debian GLIBC 2.31-9 and Python 3.9.2

**Reply by ghost** [reply, marked as answer]
I've found a bug in my code that caused this. I don't know why it throws a memory error in Linux and works fine in Windows though.

**Comment by ReemAbuRajab** [comment]
or-tool must be clone in linux only that's why i guess
